src/main.rs

```diff
@@ -1,3 +1,73 @@
+#[macro_use]
+extern crate clap;
+
 fn main() {
-    println!("Hello, world!");
+    let app = clap_app!(myapp =>
+        (version: "0.1.0")
+        (author: "Bike Barn <https://github.com/bike-barn/hermit>")
+        (about: "A home directory configuration management assistant.")
+        (@subcommand add =>
+            (about: "Add files to your hermit shell")
+            (usage: "hermit add [[-r|--redact=] <filepattern>…]")
+        )
+        (@subcommand clone =>
+            (about: "Create a local shell from an existing remote shell")
+            (usage: "hermit clone <repository> [shell-name]")
+        )
+        (@subcommand doctor =>
+            (about: "Make sure your hermit setup is sane")
+            (usage: "hermit doctor")
+        )
+        (@subcommand git =>
+            (about: "Run git operations on the current shell")
+            (usage: "hermit git <git arguments>")
+        )
+        (@subcommand init =>
+            (about: "Create a new hermit shell")
+            (usage: "hermit init [shell-name]")
+        )
+        (@subcommand nuke =>
+            (about: "Permanently remove a hermit shell")
+            (usage: "hermit nuke <shell-name>")
+        )
+        (@subcommand status =>
+            (about: "Display the status of your hermit shell")
+            (usage: "hermit status")
+        )
+        (@subcommand use =>
+            (about: "Switch to using a different hermit shell")
+            (usage: "hermit use [shell-name]")
```

Comment:
Shouldn't the shell-name be required here? or did we have a "default" shell name fallback?